doc/bgp_pic/bgp_pic.md

```diff
@@ -0,0 +1,200 @@
+<!-- omit in toc -->
+# BGP PIC HLD
+<!-- omit in toc -->
+### Revision
+| Rev |     Date    |       Author           | Change Description                |
+|:---:|:-----------:|:----------------------:|-----------------------------------|
+| 0.1 | Oct  8 2023 |   Eddie Ruan / lingyu Zhang   | Initial Draft              |
+
+<!-- omit in toc -->
+## Table of Content
+- [Goal \& Scope](#goal--scope)
+
+## Goal and Scope
+BGP PIC, as detailed in the RFC available at https://datatracker.ietf.org/doc/draft-ietf-rtgwg-bgp-pic/, addresses the enhancement of BGP route convergence. This document outlines a method to arrange forwarding structures that can lead to improved BGP route convergence. BGP PIC offers two primary enhancements:
+
+1. It prevents BGP load balancing updates from being triggered by IGP load balancing updates. This issue, which is discussed in the SONiC Routing Working Group (https://lists.sonicfoundation.dev/g/sonic-wg-routing/files/SRv6%20use%20case%20-%20Routing%20WG.pptx), can be effectively resolved using BGP PIC.
+
+<figure align=center>
+    <img src="images/srv6_igp2bgp.jpg" >
+    <figcaption>Figure 1. Alibaba issue Underlay routes flap affecting Overlay SRv6 routes <figcaption>
+</figure> 
+Note: we only handle VPN overlay routes via BGP PIC. For global table's recursive routes handling, it would be handled via a seperate HLD and done by Accton team.
+
+2. We aim to achieve fast convergence in the event of a hardware forwarding failure related to a remote BGP PE becoming unreachable. Convergence in the slow path forwarding mode is not a priority.
+
+## High Level Design
+One of the challenges in implementing PIC within FRR is the absence of PIC support in the Linux kernel. To minimize alterations in FRR while enabling PIC on platforms that do not require Linux kernel support for this feature, we are primarily focused on two key modifications:
+1. In the 'zebra' component:
+    - Introduce a new Next Hop Group (PIC-NHG) specifically for the FORWARDING function. This NHG will serve as the shareable NHG in hardware.
+    - When a BGP next hop becomes unavailable, zebra will first update the new FORWARDING-ONLY NHG before BGP convergence takes place.
+    - If changes occur in the IGP NHG and these changes do not affect the reachability of individual members within the BGP NHG, there is no need to update the BGP NHG.
+    - Zebra will transmit two new forwarding objects, BGP PIC context, and NHG, to orchagent via FPM. The handling of NHG is outlined in https://github.com/sonic-net/SONiC/pull/1425.
+    - Zebra will continue to update kernel routes in the same manner as before, as the kernel does not support BGP PIC.
```

Comment:
Is it possible to have the same behaviour on FPM and on local kernel ?
I understand that BGP PIC context is a nexthop with its resolution context, ready to use. So what about having kernel using it ? (as you said, "When a BGP next hop becomes unavailable, zebra will first update the new FORWARDING-ONLY NHG before BGP convergence takes place").

Reply:
Do you add some kind of PIC support in kernel?

Reply:
I think if you could clarify what contains BGP PIC NHG, this would help understand what needs to be done for kernel.
Today, in kernel
```
# ip nexthop ls
nexthop encap segs 2001::1
      via 172:31::2:4 dev r1-eth3
      via 172:31::2:2 dev r1-eth2
nexthop encap segs 2001::2
      via 172:31::2:4 dev r1-eth3
      via 172:31::2:2 dev r1-eth2
```
If 172:31::2:4 is unreachable, then the nexthops should be updated.
And I think today, only BGP knows how to do this.
By PIC support I expect
- first some trigger in zebra to handle this change, before BGP takes the hand
whenever the IGP resolution changes, the nexthop(s) defined for all the routes should be updated too.
- maybe a linked-list or some extra context could be useful to know which contexts should be modified.( that is why I was thinking of BGP PIC context, but perhaps it is a wrong example).

Reply:
In your example, it is current SRv6 IGP load balance struct.

With PIC support, we could have
PIC_NHG be